src/lein_monolith/plugin.clj

```diff
@@ -152,23 +175,33 @@
 
 ;; ## Plugin Middleware
 
+(defn- add-profiles
+  "Adds profiles to the project. Profiles should be passed in as a vector to ensure profiles are
+   added in the right order."
+  [project profiles]
+  (reduce (fn [project [profile-key profile]]
+            (-> project
+                (project/add-profiles {profile-key profile})
+                (project/merge-profiles [profile-key])))
+          project
+          profiles))
```

Comment:
These can both be done as batch steps:
```suggestion
  (-> project
      (project/add-profiles (into {} profiles))
      (project/merge-profiles (mapv first profiles))))
```
Especially since each call to `merge-profiles` invokes [`set-profiles`](https://github.com/technomancy/leiningen/blob/135724235b1a963a806e24d036326a1bddeced1a/leiningen-core/src/leiningen/core/project.clj#L964-L971), which will load plugins and apply middleware, we only want to call this once.